File: server/internal/models/company_model.go
package models

import (
	"time"
)

type (
	RegisterCompanyRequest struct {
		Email       string `json:"email" validate:"required"`
		Password    string `json:"password" validate:"required"`
		Name        string `json:"name" validate:"required"`
		Description string `json:"description"`
		PhoneNumber string `json:"phone_number" validate:"required"`
	}

	LoginCompanyRequest struct {
		Email    string `json:"email" validate:"required"`
		Password string `json:"password" validate:"required"`
	}

	CompanyIDRequest struct {
		ID int `json:"id" validate:"required"`
	}

	UpdateCompanyRequest struct {
		Email       string `json:"email" validate:"required"`
		Password    string `json:"password" validate:"required"`
		Name        string `json:"name" validate:"required"`
		Description string `json:"description"`
		PhoneNumber string `json:"phone_number" validate:"required"`
	}
)

type (
	CompanyResponse struct {
		ID           int                         `json:"id"`
		Email        string                      `json:"email"`
		Name         string                      `json:"name"`
		Description  string                      `json:"description"`
		PhoneNumber  string                      `json:"phone_number"`
		Address      string                      `json:"column:address"`
		// JobVacancies []models.JobVacancyResponse `json:"job_vacancies"`

		Token        string    `json:"token"`
		RefreshToken string    `json:"refresh_token"`
		CreatedAt    time.Time `json:"created_at"`
		UpdatedAt    time.Time `json:"updated_at"`
	}
)

File: lib/data/model/company_model.dart
import 'package:equatable/equatable.dart';
import 'package:json_annotation/json_annotation.dart';

part 'company_model.g.dart';

@JsonSerializable()
class CompanyModel extends Equatable {
  final int id;
  final String email;
  final String name;
  @JsonKey(name: 'phone_number')
  final String phoneNumber;
  final String address;
  final String description;
  final String token;
  @JsonKey(name: 'refresh_token')
  final String refreshToken;
  @JsonKey(name: 'created_at')
  final DateTime createdAt;
  @JsonKey(name: 'updated_at')
  final DateTime updatedAt;

  CompanyModel({
    required this.id,
    required this.email,
    required this.name,
    required this.phoneNumber,
    required this.address,
    required this.description,
    required this.token,
    required this.refreshToken,
    required this.createdAt,
    required this.updatedAt,
  });

  factory CompanyModel.fromJson(Map<String, dynamic> json) => _$CompanyModelFromJson(json);
  Map<String, dynamic> toJson() => _$CompanyModelToJson(this);
  
  @override
  List<Object?> get props => [
    id,
    email,
    name,
    phoneNumber,
    address,
    description,
    token,
    refreshToken,
    createdAt,
    updatedAt
  ];
}

@JsonSerializable()
class CompanyRegisterRequest {
  final String email;
  final String password;
  final String name;
  @JsonKey(name: 'phone_number')
  final String phoneNumber;
  final String address;
  final String description;

  CompanyRegisterRequest({
    required this.email,
    required this.password,
    required this.name,
    required this.phoneNumber,
    required this.address,
    required this.description,
  });

  Map<String, dynamic> toJson() => _$CompanyRegisterRequestToJson(this);
}

@JsonSerializable()
class CompanyLoginRequest {
  final String email;
  final String password;

  CompanyLoginRequest({
    required this.email,
    required this.password,
  });

  Map<String, dynamic> toJson() => _$CompanyLoginRequestToJson(this);
}